### PR TITLE
Support choosing a notification action with 'makoctl menu'

### LINKED ISSUE #4: Support choosing a notification action
Do we want to use dmenu for this?

### PR DESCRIPTION
**Pull request by xPMo**
Requires `jq`, alternative to #194. Would close #4.

### CODE REVIEW

File: makoctl [hunk 1]
```diff
@@ -53,6 +55,30 @@ case "$1" in
...
+	shift 1
+	if ! type -p jq > /dev/null; then
+		echo >&2 "$0: jq is required to use 'menu'"
+		exit 1
+	fi
+	if array="$(call ListNotifications | jq -re '.data[0][0].actions.data | if length > 0 then . else false end')"; then
+		sel="$(jq -rn "$array|values[]" | "$@")"
+		sel="$(jq -rn --arg sel "$sel" "$array|to_entries[]|select(.value == \$sel).key")"
+		if [ -n "$sel" ]; then
+			# The key may contain C escape sequences, use printf %b to evaluate them.
```
Comment (emersion) [inline makoctl]: Why would the key contain escape sequences that need to be evaluated?

Comment (xPMo) [reply makoctl]: It won't, my mistake. We will still need to add a trailing character to prevent newlines from being mangled.

Comment (emersion) [reply makoctl]: Honestly it's probably not worth it. Action identifiers shouldn't contain newlines.

Comment (xPMo) [reply makoctl]: Fair.

Comment (xPMo) [reply makoctl]: Yeah, the odds of encountering it is ridiculously small. I am inclined to guard against a possible future issue, but I'll concede.

**Comment by xPMo** [comment]
Slight tangent; it looks like Drew maintains `mako-git` on the AUR. Could you ask him to add `optdepends=('jq: support for makoctl menu')` to the PKGBUILD?

**Comment by emersion** [comment]
cc @ddevault ^

**Comment by xPMo** [comment]
[Actually, I went ahead and took the opportunity to use `git send-email`](https://lists.sr.ht/~sircmpwn/public-inbox/%3C20191027190435.141513-1-GammaFunction%40vivaldi.net%3E)

**Comment by p00f** [comment]
@emersion Is there supposed to be a "dismiss" action in the wofi menu or am I understanding incorrectly? I only see "activate"

**Comment by emersion** [comment]
No, dismiss is not an action. It's a separate makoctl command.

**Pull request merged**